core/app/models/mno/user.rb

```diff
@@ -0,0 +1,212 @@
+module Mno
+  class User < BaseResource
+    include MnoEnterprise::Concerns::Models::IntercomUser
+    extend Devise::Models
+    # include ActiveModel::Model
+    include ActiveModel::Conversion
+    include ActiveModel::AttributeMethods
+    include ActiveModel::Validations
+
+    property :id
+    property :created_at, type: :time
+    property :updated_at, type: :time
+    property :confirmed_at, type: :time
+    property :email, type: :string
+    property :name, type: :string
+    property :surname, type: :string
+    property :company, type: :string
+    property :phone, type: :string
+    property :password
+    property :api_secret, type: :string
+    property :api_key, type: :string
+    property :phone_country_code, type: :string
+    property :geo_country_code, type: :string
+    property :website, type: :string
+    property :sso_session, type: :string
+    property :admin_role, type: :string
+    property :avatar_url, type: :string
+
+    property :locked_at, type: :time
+    property :last_sign_in_ip
+
+    define_model_callbacks :validation #required by Devise
+    define_model_callbacks :update #required by Devise
+    define_model_callbacks :create #required by Devise
+    define_model_callbacks :save #required by Devise
+
+
+    class RemoteUniquenessValidator < ::ActiveModel::EachValidator
```

Comment:
Not a big fan of nested class. Can you have that in a separate class like `core/lib/her_extension/validations/remote_uniqueness_validation.rb`?